library/core/src/async_iter/async_iter.rs

```diff
@@ -134,3 +135,26 @@ impl<T> Poll<Option<T>> {
     #[cfg_attr(not(bootstrap), lang = "AsyncGenFinished")]
     pub const FINISHED: Self = Poll::Ready(None);
 }
+
+/// Convert something into an async iterator
+#[unstable(feature = "async_iterator", issue = "79024")]
+pub trait IntoAsyncIterator {
```

Comment:
Do we need to introduce this now? I'd rather not need to get T-libs-api sign off for this... but I think this would need T-libs-api sign-off.

Reply:
I'll factor it out.

Reply:
Okay, I've pulled out the `IntoAsyncIterator` changes and will push them to a second branch that I'll turn into PR and get T-libs-api signoff on.